src/home_robot/home_robot/manipulation/heuristic_pick_policy.py

```diff
@@ -193,20 +230,77 @@ def get_action(
                 action = ContinuousNavigationAction([0, 0, self.orient_turn_angle])
                 self.orient_turn_angle = 0
                 self.t_start_pick = self.timestep + 1
-                self.t_relative_back = 0
-                self.t_relative_standoff = 1
-                self.t_relative_grasp = 2
-                self.t_relative_snap_object = 3
+
+                if self.ik:
+                    self.t_relative_snap_object = 1
+                    self.t_relative_goto_ik = 0
+                else:
+                    self.t_relative_back = 0
+                    self.t_relative_standoff = 1
+                    self.t_relative_grasp = 2
+                    self.t_relative_snap_object = 3
+
         if action is not None:
             return action, vis_inputs
 
-        if self.timestep == self.t_start_pick + self.t_relative_back:
+        if self.timestep == self.t_start_pick + self.t_relative_goto_ik:
+            grasp_voxel = self.get_object_pick_point(obs, vis_inputs)
+            if grasp_voxel is not None:
+                self.grasp_voxel, vis_inputs = grasp_voxel
+            print("Grasp voxel:", grasp_voxel)
+            pt = self.grasp_voxel
+            # TODO: make thie constant
+            rot = tra.euler_matrix(0, 0, np.pi)
+            pt = trimesh.transform_points(pt[None], rot)[0]
+
+            pt[2] += STRETCH_STANDOFF_DISTANCE
```

Comment:
Is this redundant? `STRETCH_STANDOFF_DISTANCE` is being added inside `_find_grasp_cfg`

Reply:
Maybe, we can check